test/integration/smoke/test_vm_life_cycle.py

```diff
@@ -1991,3 +1990,135 @@ def test_01_vapps_vm_cycle(self):
+            cls.apiclient,
+            cls.services["small"],
+            accountid=cls.account.name,
+            domainid=cls.account.domainid,
+            serviceofferingid=cls.small_offering.id,
+            mode=cls.services["mode"]
+        )
+        cls.medium_virtual_machine = VirtualMachine.create(
+            cls.apiclient,
+            cls.services["small"],
+            accountid=cls.account.name,
+            domainid=cls.account.domainid,
+            serviceofferingid=cls.medium_offering.id,
+            mode=cls.services["mode"]
+        )
+        cls.virtual_machine = VirtualMachine.create(
+            cls.apiclient,
+            cls.services["small"],
+            accountid=cls.account.name,
+            domainid=cls.account.domainid,
+            serviceofferingid=cls.small_offering.id,
+            mode=cls.services["mode"]
+        )
+        cls._cleanup = [
+            cls.small_offering,
+            cls.medium_offering,
+            cls.account
+        ]
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestCloneVM, cls).tearDownClass()
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            # Clean up, terminate the created ISOs
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags = ["clone","devcloud", "advanced", "smoke", "basic", "sg"], required_hardware="false")
+    def test_clone_vm_and_volumes(self):
+        small_disk_offering = DiskOffering.list(self.apiclient, name='Small')[0];
+        config = Configurations.list(self.apiclient,
+                              name="kvm.snapshot.enabled"
+                              )
+        if len(config) == 0 or config[0].value != "true":
+            self.skipTest("Please enable kvm.snapshot.enable global config")
+        if self.hypervisor.lower() in ["kvm", "simulator"]:
+            small_virtual_machine = VirtualMachine.create(
+                self.apiclient,
+                self.services["small"],
+                accountid=self.account.name,
+                domainid=self.account.domainid,
+                serviceofferingid=self.small_offering.id,)
+            vol1 = Volume.create(
+                self.apiclient,
+                self.services,
+                account=self.account.name,
+                diskofferingid=small_disk_offering.id,
+                domainid=self.account.domainid,
+                zoneid=self.zone.id
+            )
+            small_virtual_machine.attach_volume(self.apiclient, vol1)
+            self.debug("Clone VM - ID: %s" % small_virtual_machine.id)
+            try:
+                clone_response = small_virtual_machine.clone(self.apiclient, small_virtual_machine)
```

Comment:
probably some cleanup item needs to extracted from this response as well.

Reply:
this is not what I mean, unless the clone_response *is* the resulting VM, I didn't check.

Reply:
checked, it is, should work as meant, thnx